lib/usdUfe/ufe/StagesSubject.cpp

```diff
@@ -395,7 +395,14 @@ void StagesSubject::stageChanged(
                     notifyWithoutExceptions<Ufe::Transform3d>(ufePath);
                 }
             }
+            // Special case when Undoing a visibility change, the notice.GetChangedInfoOnlyPaths()
+            // does not contain the change, hence handling visibility notification in re-synch path.
+            if (changedPath.GetNameToken() == UsdGeomTokens->visibility) {
+                Ufe::VisibilityChanged vis(ufePath);
+                notifyWithoutExceptions<Ufe::Object3d>(vis);
+            }
```

Comment:
I'm not sure the order matters, but I wonder if sending this extra visibilityChanged should come after the processAttributeChanges() just below. This would then match the order in the ChangedInfoOnlyPaths() loop.

Reply:
@seando-adsk  I moved it as per [this finding](https://github.com/Autodesk/maya-usd/pull/3345#issuecomment-1734439926) since having it in at line 398 caused duplicate notifications to be sent which failed the visibility test.